textcursor.go

```diff
@@ -0,0 +1,157 @@
+// Copyright 2015 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package gxui
+
+import "github.com/google/gxui/interval"
+
+// TextCursor represents a text position and selection. A selection is defined as
+// the pair of cursor indices sorted least to greatest. When transforming a
+// cursor with no selection, mark should equal index.
+type TextCursor struct {
```

Comment:
How about, instead of trying to treat this thing as two indices, we have:

```go
type TextCursor struct {
  Index uint // Carat index.
  Length int // If non-zero cursor is a selection.
}
```
Where Length can be zero, positive or negative.